packages/atproto_client/models/string_formats.py

```diff
@@ -0,0 +1,247 @@
+import re
+from datetime import datetime
+from typing import Callable, Mapping, Set, Union
+from urllib.parse import urlparse
+
+from pydantic import BeforeValidator, Field, ValidationInfo
+from pydantic_core import core_schema
+from typing_extensions import Annotated, Literal
+
+_OPT_IN_KEY: Literal['strict_string_format'] = 'strict_string_format'
+
+# constants
+MAX_HANDLE_LENGTH: int = 253
+MAX_NSID_LENGTH: int = 317
+MAX_RECORD_KEY_LENGTH: int = 512
+MAX_URI_LENGTH: int = 8 * 1024
+MIN_CID_LENGTH: int = 8
+TID_LENGTH: int = 13
+INVALID_RECORD_KEYS: Set[str] = {'.', '..'}
+MAX_DID_LENGTH: int = 2048  # Method-specific identifier max length
+MAX_AT_URI_LENGTH: int = 8 * 1024
+
+# patterns
+DOMAIN_RE = re.compile(r'^([a-zA-Z0-9]([a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?\.)+[a-zA-Z][a-zA-Z0-9-]*[a-zA-Z]$')
+DID_RE = re.compile(
+    r'^did:'  # Required prefix
+    r'[a-z]+:'  # method-name (lowercase only)
+    r'[a-zA-Z0-9._%-]{1,2048}'  # method-specific-id with length limit
+    r'(?<!:)$'  # Cannot end with colon
+)
+NSID_RE = re.compile(
+    r'(?![0-9])'  # Can't start with number
+    r'((?!-)[a-z0-9-]{1,63}(?<!-)\.){2,}'  # At least 2 segments, each 1-63 chars
+    r'[a-zA-Z]'  # Last segment must start with letter
+    r'[a-zA-Z0-9-]*'  # Middle chars
+    r'[a-zA-Z]'  # Must end with letter
+    r'$'  # End of string
+)
+LANG_RE = re.compile(r'^(i|[a-z]{2,3})(-[A-Za-z0-9-]+)?$')
+RKEY_RE = re.compile(r'^[A-Za-z0-9._:~-]{1,512}$')
+TID_RE = re.compile(rf'^[2-7a-z]{{{TID_LENGTH}}}$')
+CID_RE = re.compile(r'^[A-Za-z0-9+]{8,}$')
+AT_URI_RE = re.compile(
```

Comment:
sorry for the long response! i am busy at work :(

as I can see this regex is different to official one: 

https://github.com/bluesky-social/atproto/blob/53fcc2fbcbda9b325eeffb5705126831baf2a427/packages/syntax/src/aturi.ts#L5)

We use official in our AtUri class:

https://github.com/MarshalX/atproto/blob/e011bb4c18e85967a90a0271ea384972d1ae8292/packages/atproto_core/uri/uri.py#L9

Probably, your regex come from re-impl mentioned in the issues. But it would be awesome to align with official one! And as I can see you can get everything from the `syntax` package! https://github.com/bluesky-social/atproto/tree/53fcc2fbcbda9b325eeffb5705126831baf2a427/packages/syntax/src. With it all the tests should pass and no need for excluding test cases

Reply:
> sorry for the long response! i am busy at work :(

no worries! so am I 🙂 

ack! I missed that - thank you!

I will update the regexs to follow those as soon as I can 👍